File: object-oriented/polymorphism/runtime/cat.dart
void main() {
  var cat = Cat();
  cat.walk();
}

class Animal {
  void walk() {
    print('Animal walk!');
  }
}

class Cat extends Animal {
  @override
  void walk() {
    super.walk();
    print('Yeay! Cat walked!');
  }
}

File: object-oriented/polymorphism/runtime/cat.kt
fun main() {
  val cat = Cat()
  cat.walk()
}

open class Animal {
  open fun walk() {
    println("${javaClass.simpleName} walk!")
  }
}

class Cat : Animal(){
  override fun walk() {
    super.walk()
    println("Yeay! ${javaClass.simpleName} walked!")
  }
}
